pkg/oauth/interactive.go

```diff
@@ -74,7 +90,40 @@ const (
 		<script>
 			document.getElementById("favicon").setAttribute("href", "data:image/svg+xml," + encodeURIComponent(document.getElementById("logo").outerHTML));
 		</script>
+
+		{{ if .Autoclose -}}
+		<script>
+			var timeout = {{ .Timeout }};
+			setTimeout(function() { this.close(); }, timeout*1000);
+			setInterval(function() {
+				timeout--;
+				document.getElementsByName("autoclose")[0].innerHTML = "This page will close automatically in " + timeout + " seconds...";
+				if (timeout <= 0) {
+					document.getElementsByName("autoclose")[0].innerHTML = "This page will close now, thank you!";
+				}
+			}, 1000);
+		</script>
+		{{- end }}
 	</body>
 </html>
 `
-)
+	// Parse the template
+	tmpl, err := template.New("success").Parse(successTemplate)
+	if err != nil {
+		log.Panicf("error parsing success template: %v", err)
```

Comment:
Is there another way we can structure this to avoid the panics? Return an error?

Reply:
Sure thing, let me look over how errors are handled elsewhere and do similar.

Reply:
@haydentherapper I took a look at how I might accomplish this but was slightly deterred from refactoring it to return an error because I wasn't sure how it could be handled in flow.go with creating an init function to call the `GetInteractiveSuccessHTML` to get the html. I'll make a commit with that kind of change and you can let me know if you want to move forward like that or revert to the previous state.